infra/util/Crc.hpp

```diff
@@ -0,0 +1,176 @@
+#ifndef INFRA_CRC_HPP
+#define INFRA_CRC_HPP
+
+#include "infra/util/ByteRange.hpp"
+#include <climits>
+#include <cstddef>
+#include <cstdint>
+
+namespace infra
+{
+    /// Generic class template for different CRC calculations.
+    ///
+    /// Some ideas borrowed from http://www.sunshine2k.de/articles/coding/crc/understanding_crc.html
+    /// Also useful:
+    /// - http://www.sunshine2k.de/coding/javascript/crc/crc_js.html
+    /// - https://crccalc.com/
+
+    template<typename CRC_TYPE>
+    constexpr CRC_TYPE PolyReverse(CRC_TYPE Polynomial)
+    {
+        CRC_TYPE ret = 0;
+        CRC_TYPE maskIn = 1;
+        CRC_TYPE maskOut = 1 << (sizeof(CRC_TYPE) * 8 - 1);
+        while (maskIn)
+        {
+            if (Polynomial & maskIn)
+                ret |= maskOut;
+            maskOut >>= 1;
+            maskIn <<= 1;
+        }
+        return ret;
+    }
+
+    template<typename CRC_TYPE, CRC_TYPE Polynomial, CRC_TYPE InitValue = 0, CRC_TYPE FinalXor = 0, bool ReflectInput = false, bool ReflectOutput = false>
+    class Crc
+    {
+        static_assert(ReflectInput == ReflectOutput, "Currently, non-matching ReflectInput and ReflectOutput is not supported");
+
+    public:
+        Crc() = default;
+        ~Crc() = default;
+
+        /// Update the CRC calculation with the next input byte.
```

Comment:
There are a lot of comments, but hardly any of them provide actual added value. Can you remove the ones that only use more words to say the same?